controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -65,47 +97,460 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
 	if len(storageClusterList.Items) > 1 {
 		return reconcile.Result{}, fmt.Errorf("multiple StorageCluster found")
 	}
+	r.storageCluster = &storageClusterList.Items[0]
+
+	r.storageConsumer = &ocsv1alpha1.StorageConsumer{}
+	if err := r.get(r.storageConsumer); err != nil {
+		r.log.Error(err, "Failed to get StorageConsumer.")
+		return reconcile.Result{}, err
+	}
 
 	var result reconcile.Result
 	var reconcileError error
-	if storagecluster.IsOCSConsumerMode(&storageClusterList.Items[0]) {
-		result, reconcileError = r.reconcileConsumerPhases(ctx, &storageClusterList.Items[0])
+	if storagecluster.IsOCSConsumerMode(r.storageCluster) {
+		result, reconcileError = r.reconcileConsumerPhases(ctx, r.storageCluster)
 	} else {
-		result, reconcileError = r.reconcileProviderPhases(ctx, &storageClusterList.Items[0])
-	}
-
-	// Apply status changes to the StorageClassClaim
-	statusError := r.Client.Status().Update(ctx, claim)
-	if statusError != nil {
-		log.Info("Failed to update StorageClassClaim status.")
+		result, reconcileError = r.reconcileProviderPhases()
 	}
 
 	// Reconcile errors have higher priority than status update errors
 	if reconcileError != nil {
 		return result, reconcileError
 	}
 
+	// Apply status changes to the StorageClassClaim
+	statusError := r.Client.Status().Update(ctx, r.storageClassClaim)
 	if statusError != nil {
 		return result, statusError
 	}
 
 	return result, nil
 }
 
+func (r *StorageClassClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&ocsv1alpha1.StorageClassClaim{}).
+		Complete(r)
+}
+
+func (r *StorageClassClaimReconciler) initReconciler(request reconcile.Request) {
+	r.namespace = request.Namespace
+
+	r.storageConsumer = &ocsv1alpha1.StorageConsumer{}
+
+	r.storageClassClaim = &ocsv1alpha1.StorageClassClaim{}
+	r.storageClassClaim.Name = request.Name
+	r.storageClassClaim.Namespace = r.namespace
+
+	if r.storageClassClaim.Spec.Type == "blockpool" {
+		r.cephBlockPool = &rookCephv1.CephBlockPool{}
+		r.cephBlockPool.Name = fmt.Sprintf("cephblockpool-%s", r.storageConsumer.Name)
```

Comment:
This will not work as at this point `r.storageConsumer.Name` is empty (see comment above)

Reply:
ack